Python/Product/PythonTools/SharedProject/Debugger/ProvideDebugAdapterAttribute.cs

```diff
@@ -0,0 +1,129 @@
+            engineKey.SetValue("AdapterLauncher", _adapterLauncherCLSID);
+
+            /*
+             * Set to "1" if the debug adapter will use the VS "Exception Setting" tool window.  The debug adapter's must
+             * support one of the following:
+             *     -Exception Breakpoints
+             *         The debug adapter's response to the "initialize" request must contain a set of ExceptionBreakpointFilters,
+             *         and the "ExceptionBreakpointCategory" property must be defined below.An optional set of
+             *         "ExceptionBreakpointMappings" may also be provided if the VS exception names do not correspond to the
+             *         "Label" properties of the ExceptionBreakpointFilters.
+             *     -Exception Options
+             *         The debug adapter's response to the "initialize" request must contain the "SupportsExceptionOptions"
+             *         and "SupportsExceptionDetailsRequest" flags, and ExceptionCategoryMapping information must be supplied.
+             */
+            engineKey.SetValue("Exceptions", 1);
+
+            /*
+             * Set to "1" if the debug adapter supports the VS exception conditions experience(For skipping exceptions in specific modules).
+             * The debug adapter's response to the "initialize" request must contain the "SupportsExceptionConditions"
+             * and "SupportsExceptionDetailsRequest" flags.
+             */
+            engineKey.SetValue("ExceptionConditions", 0);
+
+            /*
+             * Debug Adapter Host settings
+             * These settings control the behavior of the Debug Adapter Host
+             *
+             * Name of the debug adapter
+             * This appears in VS in several places.  For example:
+             *     -The "Select Code Type" dialog for choosing which debugger to attach to a process(if Attach is supported)
+             *     -The "Debugging" column of the "Processes" tool window
+             */
+            engineKey.SetValue("Name", _name);
+
+            /*
+             * Path to the debug adapter executable
+             */
+            engineKey.SetValue("Adapter", @"$PackageFolder$\DebugAdapter.exe");
+
+            /*
+             * Arguments for the debug adapter executable (optional)
+             */
+            // engineKey.SetValue("AdapterArgs", "");
+
+            /*
+             * Language name
+             * This appears in (e.g.) the "Language" column of the Stack Trace tool window.
+             */
+            engineKey.SetValue("Language", _languageName);
+            engineKey.SetValue("LanguageId", _languageId);
+
+            /* 
+             * Adapter launcher registration 
+             */
+            var adapterKey = context.CreateKey($"CLSID\\{_adapterLauncherCLSID}");
+            adapterKey.SetValue("Assembly", "Microsoft.PythonTools.Debugger");
```

Comment:
Might be a good idea to use `typeof(...).FullName` here, to make this refactoring-proof.